packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3235 @@
+  }
+
+  @override
+  List<MenuItem> get descendants => PlatformMenu.getDescendants(this);
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return <DiagnosticsNode>[
+      ...menus.map<DiagnosticsNode>((MenuItem child) {
+        return child.toDiagnosticsNode();
+      })
+    ];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<Widget>('leadingIcon', leadingIcon, defaultValue: null));
+    properties.add(StringProperty('label', label));
+    properties.add(DiagnosticsProperty<Widget>('trailingIcon', trailingIcon, defaultValue: null));
+    properties.add(StringProperty('semanticLabel', semanticLabel, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<ShapeBorder?>>('shape', shape, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<double?>>('elevation', elevation, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('padding', padding, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('buttonPadding', buttonPadding, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonBackgroundColor', buttonBackgroundColor,
+        defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonForegroundColor', buttonForegroundColor,
+        defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonOverlayColor', buttonOverlayColor,
+        defaultValue: null));
+    properties
+        .add(DiagnosticsProperty<MaterialStateProperty<ShapeBorder?>>('buttonShape', buttonShape, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<TextStyle?>>('buttonTextStyle', buttonTextStyle, defaultValue: null));
+  }
+}
+
+class _MenuBarMenuState extends State<MenuBarMenu> {
+  _MenuNode? menu;
+  _MenuBarController? controller;
+  bool registered = false;
+  Timer? hoverTimer;
+  Timer? clickBanTimer;
+  bool clickBan = false;
+  late FocusNode focusNode;
+
+  bool get _isAnOpenMenu {
+    return menu != null && controller!.isAnOpenMenu(menu!);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    focusNode = FocusNode(debugLabel: widget.label);
+    focusNode.addListener(_markDirty);
+  }
+
+  @override
+  void dispose() {
+    focusNode.removeListener(_markDirty);
+    focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    _updateChangedState();
+    super.didChangeDependencies();
+  }
+
+  @override
+  void didUpdateWidget(MenuBarMenu oldWidget) {
+    _updateChangedState();
```

Comment:
That's a good point.  OK, removed that from here, and now I just do the equivalent in `didChangeDependencies`, so that we can catch when the controller or node changes.